documentation/developer_setup.md

```diff
@@ -100,33 +79,41 @@ tsp-client bundles up the client generation process into a single program, makin
+1. Add the go configuration to your TypeSpec project in the `tspconfig.yaml` file in azure-rest-api-specs: ([example](https://github.com/Azure/azure-rest-api-specs/blob/bd235f0c4ef6b3887dae6658a0a3a766a6fa4887/specification/eventgrid/Azure.Messaging.EventGrid/tspconfig.yaml#L57)).
 
 	```yaml
 	# other YAML elided
 	options:
 	  # other emitters elided
 	  "@azure-tools/typespec-go":
-        module: "github.com/Azure/azure-sdk-for-go/sdk/<path to your 'az' package goes here>"
-        module-version: "0.0.1"
-        emitter-output-dir: "{project-root}"
+		module: "github.com/Azure/azure-sdk-for-go/{service-dir}/aznamespaces"
+		service-dir: "sdk/messaging/eventgrid"
+		emitter-output-dir: "{output-dir}/{service-dir}/aznamespaces"
 	```
 2. Create a tsp-location.yaml file at the root of your module directory. This file gives the location and commit that should be used to generate your code: ([example](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/messaging/eventgrid/aznamespaces/tsp-location.yaml)).
+	``` yaml
+	directory: specification/eventgrid/Azure.Messaging.EventGrid
+	commit: 8d6deb81acb126a071f6f7dbf18d87a49a82e7e2
+	repo: Azure/azure-rest-api-specs
+	```
+
+3. Install [`tsp-client`](https://github.com/Azure/azure-sdk-tools/blob/main/tools/tsp-client/README.md). If already installed, be sure to update to the latest version.
+4. In a terminal, `cd` into your package folder and type `tsp-client update`. This should run without error and will create a client, along with code needed to serialize and deserialize models.
+
+	```shell
+	azure-sdk-for-go/sdk/messaging/eventgrid/aznamespaces$ tsp-client update
+	```
 
-3. Install `tsp-client`: [(instructions)](https://github.com/Azure/azure-sdk-tools/blob/main/tools/tsp-client/README.md)
-4. Create a `build.go` file, and add `go generate` commands to run tsp-client: ([example](https://github.com/Azure/azure-sdk-for-go/blob/656c879ce93a0bc011d60df1f7b300620be08f82/sdk/messaging/eventgrid/aznamespaces/build.go#L4)).
-    
-	NOTE: the formatting here is _very_ important:
-	```go
-    //go:generate tsp-client update
-    //go:generate goimports -w .
+	To generate using a local TypeSpec project,
+	```shell
+	azure-sdk-for-go/sdk/messaging/eventgrid/aznamespaces$ tsp-client update --local-spec-repo <path to TypeSpec project>
 	```
 
-4. In a terminal, `cd` into your package folder and type `go generate`. This should run without error and will create a client, along with code needed to serialize and deserialize models.
+Generated code **must not** be edited, as any edits would be lost on future regeneration of content. To make customizations, update the TypeSpec project's `client.tsp` file then regenerate.
 
 ### Using Autorest
```

Comment:
Lets add ** DEPRECATED **

Reply:
added in [38c9ae3](https://github.com/Azure/azure-sdk-for-go/pull/25295/commits/38c9ae3229895758a7ba0462ef33aac79f573ed0)